consensus/src/liveness/leader_reputation.rs

```diff
@@ -90,40 +90,61 @@ impl AptosDBBackend {
 
     fn get_from_db_result(
         &self,
+        target_epoch: u64,
         target_round: Round,
         events: &Vec<NewBlockEvent>,
         hit_end: bool,
     ) -> Vec<NewBlockEvent> {
+        let has_larger = events.first().map_or(false, |e| {
+            (e.epoch(), e.round()) >= (target_epoch, target_round)
+        });
+        if !has_larger {
+            // error, and not a fatal, in an unlikely scenario that we have many failed consecutive rounds,
+            // and nobody has any newer successful blocks.
+            error!(
```

Comment:
we should sample both error logs, they're very spammy when it's wrong

Reply:
Let me land this, and will sync up with you for the sampling.

if these show up, something is very wrong.